### PR TITLE
ruby: Add pioinfo discovery in aarch64

### PR DESCRIPTION
**Pull request by hmartinez82**

### CODE REVIEW

File: mingw-w64-ruby/PKGBUILD [hunk 1]
```diff
@@ -10,7 +10,7 @@ _realname=ruby
 pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 pkgver=3.1.3
-pkgrel=2
+pkgrel=3
 pkgdesc="An object-oriented language for quick and easy programming (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32')
```
Comment (mmuetzel) [inline mingw-w64-ruby/PKGBUILD:16]: Does it build and work for that platform?
``` suggestion
mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
```

Comment (hmartinez82) [reply mingw-w64-ruby/PKGBUILD:16]: That's why still a draft. I wanted the CI to do its thing to make sure I didn't break anything else while I ran the build in an CLANGARM64 env. 
It seems it did not work :(
```
9 warnings generated.
45 warnings generated.
11 warnings generated.
linking miniruby.exe
make: *** [uncommon.mk:852: .rbconfig.time] Segmentation fault
make: *** Waiting for unfinished jobs....
make: *** [uncommon.mk:1190: builtin_binary.inc] Segmentation fault
==> ERROR: A failure occurred in build().
    Aborting...
```

I used MSVC on ARM64 to debug the ARM64 and look at disassembly code. Was that the wrong approach?

Comment (hmartinez82) [reply mingw-w64-ruby/PKGBUILD:16]: @mmuetzel Just a heads up. That failure in CLANG32 and CLANG64 is happening even in the master branch. At least I tried the master branch locally in CLANG64 and it fails in the same step.

**Comment by lazka** [comment]
See https://github.com/ruby/ruby/pull/8995

**Comment by hmartinez82** [comment]
> See [ruby/ruby#8995](https://github.com/ruby/ruby/pull/8995)

Awesome, thanks. I also found a bug when I was porting my fix from a local project to debug this in MSVC to the diff

**Comment by hmartinez82** [comment]
Omg, we literally went thru the same bit acrobatics 12 hours apart 😑😅
His PR is way ahead of mine because it has further works. I'll close this for now. Thanks @lazka

**Comment by lazka** [comment]
At first I was assuming you copied it, but then saw the difference yeah :D

**Comment by hmartinez82** [comment]
It was a lot of pain lol. I haven't done this much bit shifting since college.

**Comment by lazka** [comment]
#19157

**Comment by pbo-linaro** [comment]
It's amazing we did it at the same time @hmartinez82!
I used this small project and script to experiment: https://github.com/pbo-linaro/pioinfo.

**Comment by pbo-linaro** [comment]
> It was a lot of pain lol. I haven't done this much bit shifting since college.

Same for me ahah. Was a bit painful to get the right masks.

**Pull request closed without merging**